src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1003,6 +1103,59 @@ def custom_forward(*inputs):
         )
 
 
+# Adapted from https://github.com/salesforce/LAVIS/blob/main/lavis/models/blip2_models/Qformer.py#L51
+class Blip2TextEmbeddings(nn.Module):
+    """Construct the embeddings from word and position embeddings."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, padding_idx=config.pad_token_id)
+        self.position_embeddings = nn.Embedding(config.max_position_embeddings, config.hidden_size)
+
+        # self.LayerNorm is not snake-cased to stick with TensorFlow model variable name and be able to load
+        # any TensorFlow checkpoint file
+        self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
```

Comment:
shall the layernorm and dropout be moved to `Blip2QFormerModel`, to be more similar to the other Blip2 model implementations?

Reply:
I think that's fine for now